flask/flask-launcher/src/main/java/net/corda/flask/launcher/Launcher.java

```diff
@@ -152,18 +154,40 @@ final int launch(Manifest manifest, List<String> jvmArgs, List<String> javaAgent
             }
             builder.getCliArgs().addAll(args);
             beforeChildJvmStart(builder);
-            try(LockFile processLock = LockFile.acquire(cache.getPidFile(), false)) {
-                builder.getProperties().put(Flask.JvmProperties.PID_FILE, cache.getPidFile());
-                Path heartbeatAgentPath = extractedLibraries.get(
-                        manifest.getMainAttributes().getValue(Flask.ManifestAttributes.HEARTBEAT_AGENT_HASH));
-                builder.getJvmArgs().add("-javaagent:" + heartbeatAgentPath);
-                int returnCode = builder.exec();
-                afterChildJvmExit(returnCode);
-                cache.touchLibraries();
-                return returnCode;
-            } finally {
-                Files.delete(cache.getPidFile());
-            }
+            builder.getProperties().put(Flask.JvmProperties.PID_FILE, cache.getPidFile());
+            Path heartbeatAgentPath = extractedLibraries.get(
+                    manifest.getMainAttributes().getValue(Flask.ManifestAttributes.HEARTBEAT_AGENT_HASH));
+            builder.getJvmArgs().add("-javaagent:" + heartbeatAgentPath);
+            LockFile processLock = LockFile.acquire(cache.getPidFile(), false);
+            Process process = builder.build().inheritIO().start();
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                if(process.isAlive()) {
+                    process.destroy();
+                    try {
+                        long timeout = Long.parseLong(
+                                System.getProperty(
+                                        Flask.JvmProperties.KILL_TIMEOUT_MILLIS,
+                                        Flask.Constants.DEFAULT_KILL_TIMEOUT_MILLIS));
+                        process.waitFor(timeout, TimeUnit.MILLISECONDS);
+                    } catch (InterruptedException ie) {
+                        log.error(ie.getMessage(), ie);
```

Comment:
This is a shutdown hook, and so the SLF4J `Logger` is probably shutting down too. It's probably better just to write to `stderr` here.